### PR TITLE
Add the release playbook to the CI repository.

### PR DESCRIPTION
**Pull request by katre**
This is a direct translation of the internal playbook, with a new edits and reformats.

There are a lot of internal links. What should we do with them?

### CODE REVIEW

File: docs/release-playbook.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+# Bazel Release Playbook
+
+Authors: lpino@, pcloudy@, laurentlb@
```
Comment (laurentlb) [inline docs/release-playbook.md]: I think we can remove the author line

File: docs/release-playbook.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+Status: Work in progress
+
+Last Updated: 2018-11-09
```
Comment (laurentlb) [inline docs/release-playbook.md]: Remove this also, history is visible on GitHub

File: docs/release-playbook.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+Link to this document: go/bazel-release-playbook
```
Comment (laurentlb) [inline docs/release-playbook.md]: Remove also

File: docs/release-playbook.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+Congratulations, you're the new release manager!
```
Comment (laurentlb) [inline docs/release-playbook.md]: Rephrase: the page should be relevant for anyone who wants to know how the release process works

File: docs/release-playbook.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+Each release has a tracking bug (see the list on http://go/bazel-release). The bug
+includes a "Target RC date". On that day, create a new release candidate.
```
Comment (laurentlb) [inline docs/release-playbook.md]: Outdated. The list of releases is here: https://github.com/bazelbuild/bazel/issues?utf8=%E2%9C%93&q=label%3Arelease+

File: docs/release-playbook.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+         should be cherry-picked, any remaining issues should become release
+         blockers.
+
+### Create a Candidate
+
+Create candidates with the release.sh script.
+
+*   If it's the first candidate for this version, run:
+
+    ```
```
Comment (laurentlb) [inline docs/release-playbook.md]: nit. add syntax highlighting

> \`\`\`bash

File: docs/release-playbook.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+        group.
+
+    *   If that worked, click "Unblock step" for the "Generate Announcement" step.
+
+*   Prepare the release announcement on http://go/bazel-newsletters.
+    *   Create a new section for the release. Populate using the generated text
+        (from the "generate announcement" step).
+    *   Reorganize the notes per category (C++, Java, etc.)
+    *   Add a comment with "+[spomorski@google.com](mailto:spomorski@google.com)" so that he takes a look.
+    *   Send an email to [blaze-devel](mailto:blaze-devel@google.com) asking for reviewers.
```
Comment (laurentlb) [inline docs/release-playbook.md]: Remove. We could send an email to everyone who has added any release note.

For now, you can just replace `blaze-devel` with `bazel-dev` (https://groups.google.com/forum/#!forum/bazel-dev)

**Comment by laurentlb** [comment]
cc @dslomov, @aehlig

File: docs/release-playbook.md (hunk 1 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 2 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 3 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 4 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 5 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 6 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 7 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    repository.
+
+    *   If you don't have the permission, ask one of the Buildkite org
+        admins to add you to the
+        [bazel-sheriffs](https://buildkite.com/bazel?team=bazel-sheriffs)
+        group.
+
+    *   If that worked, click "Unblock step" for the "Generate Announcement" step.
+
+*   Prepare the release announcement on http://go/bazel-newsletters.
```
Comment (laurentlb) [inline docs/release-playbook.md]: Expand link: https://docs.google.com/document/d/1wDvulLlj4NAlPZamdlEVFORks3YXJonCjyuQMUQEmB0/edit

File: docs/release-playbook.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+    1.  **Why?** Sometimes binaries are uploaded incorrectly.
+    1.  **How?** Go to the [GH releases page](https://github.com/bazelbuild/bazel/releases),
+        click "Edit", see if there's a red warning sign next to any binary. You
+        need to manually upload those; get them from
+        `https://storage.googleapis.com/bazel/$RELEASE_NUMBER/release/index.html`.
+1.  Update the release bug:
+    1.  State the fact that you pushed the release
+    1.  Ask the package maintainers to update the package definitions: @vbatts @petemounce
+    1.  Example: [https://github.com/bazelbuild/bazel/issues/3773#issuecomment-352692144]
+1.  Publish blog post (http://go/bazel-newsletters).
```
Comment (laurentlb) [inline docs/release-playbook.md]: same here

File: docs/release-playbook.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    *   File bugs (**TODO: how to find the owner/project link?**)
+
+*   Once issues are fixed, create a new candidate with the relevant cherry-picks.
+
+## Push a release
+
+1.  Verify that the following conditions **all apply**:
+    1.  at least **2 weeks passed since you pushed RC1**, and
+    1.  at least **2 business days passed since you pushed the last RC**, and
+    1.  there are **no open [release-critical Blaze bugs](https://b.corp.google.com/hotlists/14305)** in Buganizer, and
```
Comment (laurentlb) [inline docs/release-playbook.md]: Remove
Google-internal bugs are not relevant

File: docs/release-playbook.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+         commit. If only a few downstream jobs are failing and the issues are
+         known then this is a good baseline commit. Fixes for the known issues
+         should be cherry-picked, any remaining issues should become release
+         blockers.
+
+### Create a Candidate
+
+Create candidates with the release.sh script.
+
+*   If it's the first candidate for this version, run:
```
Comment (laurentlb) [inline docs/release-playbook.md]: nit. use ordered list? It might be easier to read or remember where I was

File: docs/release-playbook.md (hunk 8 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 9 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 10 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

File: docs/release-playbook.md (hunk 11 above)
Comment (katre) [reply docs/release-playbook.md]: Done.

**Pull request merged**